### PR TITLE
1. Add support for gcp cmk

### PR DESCRIPTION
**Pull request by rohankabra-db**
Tested acceptance test by running `bash-3.2$ scripts/run.sh gcp-accounts 'TestGcpaAccCustomerManagedKeysForStorage' --debug --tee` -
```


3 tf_rpc=ApplyResourceChange
2023-02-16T17:10:32.183-0800 [TRACE] sdk.helper_schema: Calling downstream: tf_resource_type=databricks_mws_customer_managed_keys tf_req_id=d4d9dd50-0291-96e6-a6b4-32e2ae934e85 tf_provider_addr=registry.terraform.io/hashicorp/databricks tf_rpc=ApplyResourceChange
2023/02/16 17:10:32 [INFO] Configured google-accounts auth: host=https://accounts.staging.gcp.databricks.com, account_id=9fcbb245-7c44-4522-9870-e38324104cf8, google_service_account=rohansa2-staging@databricks-staging-customer.iam.gserviceaccount.com
2023/02/16 17:10:33 [DEBUG] POST /api/2.0/accounts/9fcbb245-7c44-4522-9870-e38324104cf8/customer-managed-keys {
  "account_id": "9fcbb245-7c44-4522-9870-e38324104cf8",
  "gcp_key_info": {
    "kms_key_id": "projects/databricks-staging-customer/locations/us-central1/keyRings/cmk-qa/cryptoKeys/cmk-qa-key"
  },
  "use_cases": [
    "STORAGE"
  ]
}
2023/02/16 17:10:34 [DEBUG] 201 Created  {
  "account_id": "9fcbb245-7c44-4522-9870-e38324104cf8",
  "creation_time": 1676596234145,
  "customer_managed_key_id": "77b97eae-24eb-422e-90ed-f36f1540acc5",
  "gcp_key_info": {
    "kms_key_id": "projects/databricks-staging-customer/locations/us-central1/keyRings/cmk-qa/cryptoKeys/cmk-qa-key"
  },
  "updated_time": 1676596234145,
  "use_cases": [
    "STORAGE"
  ]
} <- POST /api/2.0/accounts/9fcbb245-7c44-4522-9870-e38324104cf8/customer-managed-keys
2023/02/16 17:10:34 [DEBUG] GET /api/2.0/accounts/9fcbb245-7c44-4522-9870-e38324104cf8/customer-managed-keys/77b97eae-24eb-422e-90ed-f36f1540acc5
2023/02/16 17:10:34 [DEBUG] 200 OK  {
  "account_id": "9fcbb245-7c44-4522-9870-e38324104cf8",
  "creation_time": 1676596234145,
  "customer_managed_key_id": "77b97eae-24eb-422e-90ed-f36f1540acc5",
  "gcp_key_info": {
    "kms_key_id": "projects/databricks-staging-customer/locations/us-central1/keyRings/cmk-qa/cryptoKeys/cmk-qa-key"
  },
  "updated_time": 1676596234145,
  "use_cases": [
    "STORAGE"
  ]
} <- GET /api/2.0/accounts/9fcbb245-7c44-4522-9870-e38324104cf8/customer-managed-keys/77b97eae-24eb-422e-90ed-f36f1540acc5
2023-02-16T17:10:34.609-0800 [TRACE] sdk.helper_schema: Called downstream: tf_provider_addr=registry.terraform.io/hashicorp/databricks tf_rpc=ApplyResourceChange tf_resource_type=databricks_mws_customer_managed_keys tf_req_id=d4d9dd50-0291-96e6-a6b4-32e2ae934e85
```

### CODE REVIEW

File: mws/resource_mws_customer_managed_keys.go [hunk 1]
```diff
@@ -17,10 +17,16 @@ type AwsKeyInfo struct {
...
+// GcpKeyInfo has information about the KMS key for BYOK
+type GcpKeyInfo struct {
+	KmsKeyId string `json:"kms_key_id"`
+}
+
 // CustomerManagedKey contains key information and metadata for BYOK for E2
 type CustomerManagedKey struct {
 	CustomerManagedKeyID string      `json:"customer_managed_key_id,omitempty" tf:"computed"`
-	AwsKeyInfo           *AwsKeyInfo `json:"aws_key_info" tf:"force_new"`
+	AwsKeyInfo           *AwsKeyInfo `json:"aws_key_info,omitempty" tf:"force_new"`
```
Comment (alexott) [inline mws/resource_mws_customer_managed_keys.go:28]: it makes sense to add `conflicts:` annotations to each of the fields, so people won't be able to specify both AWS & GCP keys

**Review by alexott — CHANGES_REQUESTED** [review]
Also, documentation for this resource wasn't updated to describe it. Also, please say that either AWS or GCP is required

File: mws/resource_mws_customer_managed_keys.go (hunk 1 above)
Comment (nfx) [reply mws/resource_mws_customer_managed_keys.go:28]: backend validates it anyway

**Comment by rohankabra-db** [comment]
@alexott @nfx The documentation [here](https://github.com/databricks/terraform-provider-databricks/blob/master/docs/resources/mws_customer_managed_keys.md) needs to be changed completely to make it cloud agnostic. At present, it just references AWS docs and AWS way of doing it.

We don't have any public facing GCP docs to share yet as the feature is still in private preview. Should we do this change later?

**Comment by alexott** [comment]
yes, it's ok to leave out documentation until the public preview

File: mws/resource_mws_customer_managed_keys.go (hunk 1 above)
Comment (rohankabra-db) [reply mws/resource_mws_customer_managed_keys.go:28]: @alexott 
Made the changes as suggested. Added conflicts to both AwsKeyInfo and GcpKeyInfo. Please verify. Will send the new revision.
```
// CustomerManagedKey contains key information and metadata for BYOK for E2
// You must specify either AwsKeyInfo for AWS or GcpKeyInfo for GCP, but not both
type CustomerManagedKey struct {
	CustomerManagedKeyID string      `json:"customer_managed_key_id,omitempty" tf:"computed"`
	AwsKeyInfo           *AwsKeyInfo `json:"aws_key_info,omitempty" tf:"force_new,conflicts:gcp_key_info"`
	GcpKeyInfo           *GcpKeyInfo `json:"gcp_key_info,omitempty" tf:"force_new,conflicts:aws_key_info"`
	AccountID            string      `json:"account_id" tf:"force_new"`
	CreationTime         int64       `json:"creation_time,omitempty" tf:"computed"`
	UseCases             []string    `json:"use_cases"`
}

```

**Comment by rohankabra-db** [comment]
@alexott Addressed your comments. Please take a look whenever you get time.

File: internal/acceptance/mws_customer_managed_keys_test.go [hunk 2]
```diff
@@ -16,3 +16,18 @@ func TestMwsAccCustomerManagedKeys(t *testing.T) {
 		}`,
 	})
 }
+
+func TestGcpaAccCustomerManagedKeysForStorage(t *testing.T) {
+	accountLevel(t, step{
+		Template: `provider "databricks" {
+				host     = "{env.DATABRICKS_HOST}"
+			}
+			resource "databricks_mws_customer_managed_keys" "this" {
```
Comment (nfx) [inline internal/acceptance/mws_customer_managed_keys_test.go:25]:
```suggestion
		Template: `resource "databricks_mws_customer_managed_keys" "this" {
```

provider doesn't have to be initialized

**Review by nfx — APPROVED** [review]

**Pull request merged**